### ISSUE TITLE
representation of the end token?

### ISSUE THREAD

**Issue by Ray0089** [JayParks/tf-seq2seq#20]
I'm a beginner in deep learning. Can i ask if the following codes are trying to add the decoder_end_token to all the decoder targets with the end_token full of 1s?
When we calculate the sequence loss, the crossentropy between the end_token and corresponding y_hat could be very large?
`       decoder_end_token = tf.ones(shape=[self.batch_size, 1], dtype=tf.int32) * data_utils.end_token 
`
`
            self.decoder_targets_train = tf.concat([self.decoder_inputs,decoder_end_token], axis=1)
`
`                self.loss = seq2seq.sequence_loss(logits=self.decoder_logits_train, 
                                                  targets=self.decoder_targets_train,
                                                  weights=masks,
                                                  average_across_timesteps=True,
                                                  average_across_batch=True,)`

**Comment by Ray0089** [comment]
I now figure out that the 1s are added to the time dimension..
